lib/dynflow/persistence_adapters/sequel.rb

```diff
@@ -58,20 +60,48 @@ def find_execution_plans(options = {})
         data_set.all.map { |record| load_data(record) }
       end
 
-      def delete_execution_plans(filters, batch_size = 1000)
+      def delete_execution_plans(filters, batch_size = 1000, backup_dir = nil)
         count = 0
+        current_dir = current_backup_dir(backup_dir) if backup_dir
         filter(:execution_plan, table(:execution_plan), filters).each_slice(batch_size) do |plans|
           uuids = plans.map { |p| p.fetch(:uuid) }
           @db.transaction do
             table(:delayed).where(execution_plan_uuid: uuids).delete
```

Comment:
The backup is done mainly for later debugging and the schedules are usually not relevant to the action run. The real execution times are stored within the plans which is sufficient.